netaddr.go

```diff
@@ -252,6 +252,75 @@ func (ip IP) IsMulticast() bool {
 	}
 }
 
+// Mask applies a CIDR mask of leading bits to IP, producing an IPPrefix
+// of the specified length. If IP is the zero value or bits is too large, a
+// zero-value IPPrefix is returned.
+func (ip IP) Mask(bits uint8) IPPrefix {
+	maxBits := uint8(32)
+	if ip.Is6() {
+		maxBits = 128
+	}
+
+	if bits > maxBits {
+		// Invalid mask size for both IPv4/6: nothing to do.
+		return IPPrefix{}
```

Comment:
Does this square with the rest of the API? Returning zero values on error is a very stdlib net kind of thing to do, but in other PRs I thought I saw that we wanted to use the standard `ret, err` pattern?

Reply:
I think I'm in favor of doing this as well actually. I had thought there were some other zero-value returns for invalid input, but it seems that most of those are around handling the zero-value IP.

For anything that accepts input, it is probably best to return errors.

Reply:
Before I change anything: assuming I add an error return, what's the proper behavior for `case nil` below? `return IPPrefix{}, nil` zero value still?

Reply:
@bradfitz was https://github.com/inetaf/netaddr/pull/16#discussion_r410464239 in response to this discussion?

Reply:
An alternative would be to panic on invalid input to the function. I don't see how an invalid bits value is an expected error, it feels to me like it would always be a programming error.

Reply:
I could see passing in a bits value that's based on user input. Panicking in that case is correct, but rude.

I'm not particularly fussed about zero value vs. errors as the return, as long as there's an easy way to check for errors (e.g. something like time.IsZero(), if we have one of those). But if a function can fail, barring extenuating circumstances I'd rather have it return errors.

Reply:
Alright, will add error returns momentarily and ask for more feedback.